src/main/java/in/projecteka/consentmanager/user/PatientsController.java

```diff
@@ -59,22 +71,99 @@
         return ReactiveSecurityContextHolder.getContext()
                 .map(securityContext -> (Caller) securityContext.getAuthentication().getPrincipal())
                 .flatMap(caller -> transactionPinService.validatePinFor(caller.getUsername(),
-                request.getPin(),
-                request.getRequestId(),
-                request.getScope()));
+                        request.getPin(),
+                        request.getRequestId(),
+                        request.getScope()));
     }
 
     @PostMapping("/profile")
     public Mono<Session> create(@RequestBody SignUpRequest request,
                                 @RequestHeader(name = "Authorization") String token) {
         var signUpRequests = SignUpRequestValidator.validate(request, userService.getUserIdSuffix());
         return signUpRequests.isValid()
-               ? Mono.justOrEmpty(signupService.sessionFrom(token))
-                       .flatMap(sessionId -> userService.create(signUpRequests.get(), sessionId)
-                               .zipWith(Mono.just(sessionId))
-                               .flatMap(tuple -> signupService.removeOf(tuple.getT2()).thenReturn(tuple.getT1())))
-               : Mono.error(new ClientError(BAD_REQUEST,
-                       new ErrorRepresentation(new Error(INVALID_REQUESTER,
-                               signUpRequests.getError().reduce((left, right) -> format("%s, %s", left, right))))));
+                ? Mono.justOrEmpty(signupService.sessionFrom(token))
+                .flatMap(sessionId -> userService.create(signUpRequests.get(), sessionId)
+                        .zipWith(Mono.just(sessionId))
+                        .flatMap(tuple -> signupService.removeOf(tuple.getT2()).thenReturn(tuple.getT1())))
+                : Mono.error(new ClientError(BAD_REQUEST,
+                new ErrorRepresentation(new Error(INVALID_REQUESTER,
+                        signUpRequests.getError().reduce((left, right) -> format("%s, %s", left, right))))));
+    }
+
+    @PostMapping("/generateotp")
+    @ResponseStatus(CREATED)
+    public Mono<GenerateOtpResponse> generateOtp(@RequestBody GenerateOtpRequest request) {
+        return profileService.profileFor(request.getUsername())
+                .switchIfEmpty(Mono.error(ClientError.userNotFound()))
+                .flatMap(this::getVerfiedIdentifier)
+                .switchIfEmpty(Mono.error(ClientError.unAuthorized()))
+                .flatMap(this::getGenerateOtpResponse)
+                .switchIfEmpty(Mono.error(ClientError.failedToGenerateOtp()));
+    }
+
+    private Mono<GenerateOtpResponse> getGenerateOtpResponse(UserSignUpEnquiry userSignUpEnquiry) {
+        SignUpSession signUpSession = userService.sendOtp(userSignUpEnquiry).block();
```

Comment:
Please don't use userService.sendOtp as it is specific to user registration. Instead, use OtpServiceClient

Reply:
Yes, there is a method added for password reset now.